resources/assets/js/volumes/cloneForm.vue

```diff
@@ -0,0 +1,190 @@
+        },
+        initializeLabelTrees(fileLabelTrees, annotationLabelTrees) {
+            const nbrTrees = fileLabelTrees.length;  // labels of fileLabelTrees and annotationLabelTrees are equal
+
+            if (this.fileLabelIds.length > 0) {
+                this.cloneFileLabels = true;
+                this.restrictFileLabels = true;
+            }
+
+            if (this.annotationLabelIds.length > 0) {
+                this.cloneAnnotationLabels = true;
+                this.restrictAnnotationLabels = true;
+            }
+
+            for (let i = 0; i < nbrTrees; i++) {
+                fileLabelTrees[i].labels.forEach((label) => this.fileLabelIds.includes(String(label.id)) ?
+                    label.selected = true : label.selected = false);
+                annotationLabelTrees[i].labels.forEach((label) => this.annotationLabelIds.includes(String(label.id)) ?
+                    label.selected = true : label.selected = false);
+            }
+
+            this.fileLabelTrees = fileLabelTrees;
+            this.annotationLabelTrees = annotationLabelTrees;
+        },
+        initializeFileList(ids, filenames) {
+            const nbrFiles = ids.length;
+            if (nbrFiles > 0) {
+                this.cloneFiles = true;
+                for (let i = 0; i < nbrFiles; i++) {
+                    this.selectedFiles.push({id: ids[i], filename: filenames[i]});
+                }
+            }
+        }
+    },
+    watch: {
+        cloneAnnotations(newState) {
+            if (!newState) {
+                this.restrictAnnotationLabels = false;
+            }
+        },
+        cloneFiles(newState) {
+            if (!newState) {
+                this.restrictFileLabels = false;
+                this.cloneAnnotationLabels = false;
+                this.restrictAnnotationLabels = false;
+            }
+        },
+        cloneFileLabels(newState) {
+            if (!newState) {
+                this.restrictFileLabels = false;
+            }
+        },
+    },
+    created() {
+        this.volume = biigle.$require('volume');
+        this.id = this.volume.id;
+        this.name = biigle.$require('name');
+        this.destinationProjects = biigle.$require('destinationProjects');
+        this.cloneUrlTemplate = biigle.$require('cloneUrlTemplate');
+        this.selectedProjectId = Number(location.href.split('=')[1]);
+        this.annotationLabelIds = JSON.parse(biigle.$require('annotationLabelIds'));
+        this.fileLabelIds = JSON.parse(biigle.$require('fileLabelIds'));
+        this.cloneFileLabels = biigle.$require('cloneFileLabels');
+        this.cloneAnnotationLabels = biigle.$require('cloneAnnotations');
+        let fileLabelTrees = JSON.parse(biigle.$require('fileLabelTrees'));
+        let annotationLabelTrees = JSON.parse(biigle.$require('annotationLabelTrees'));
+        let ids = JSON.parse(biigle.$require('selectedFilesIds'));
+        let filenames = JSON.parse(biigle.$require('selectedFiles'));
```

Comment:
See my other comment.

```suggestion
        this.annotationLabelIds = biigle.$require('annotationLabelIds');
        this.fileLabelIds = biigle.$require('fileLabelIds');
        this.cloneFileLabels = biigle.$require('cloneFileLabels');
        this.cloneAnnotationLabels = biigle.$require('cloneAnnotations');
        let fileLabelTrees = biigle.$require('fileLabelTrees');
        let annotationLabelTrees = biigle.$require('annotationLabelTrees');
        let ids = biigle.$require('selectedFilesIds');
        let filenames = biigle.$require('selectedFiles');
```